NEWS

```diff
@@ -1,5 +1,185 @@
+
+- `barman-wal-restore` now moves WALs from the spool directory to `pg_wal`
+  instead of copying them. This can improve performance if the spool directory
+  and the `pg_wal` directory are in the same partition of the filesystem.
+
+- `barman check-backup` now shows the reason why a backup was marked as `FAILED`
+  in the output and logs. Previously for a user to know why the backup was
+  marked as `FAILED`, they would need to run `barman show-backup` command.
+
+- Add configuration option `aws_await_snapshots_timeout` and the corresponding
+  `--aws-await-snapshots-timeout` command-line option on `barman-cloud-backup`.
+  This specifies the timeout in seconds to wait for snapshot backups to reach
+  the completed state.
+
+- Add a keep-alive mechanism to rsync-based backups. Previously the Postgres
+  session created by Barman to run `pg_backup_start` and `pg_backup_stop` would
+  be idle for how long the base backup copy could take. That could lead to a
+  firewall or router dropping the connection because it was idle for a long
+  time. The keep-alive mechanism keeps sending heartbeat queries to Postgres
+  through that connection, thus reducing the likelihood of a connection
+  droppage. The interval between heartbeats can be controlled through the new
+  configuration option `keepalive_interval` and through the corresponding CLI
+  option `--keepalive-interval` of the `barman backup` command.
+
+- Bug fixes:
+
+  - When recovering a backup with “no get wal” approach and `--target-time` is
+    set, copy all WAL files. Previously Barman would attempt to “guess” the WAL
+    files required by Postgres to reach the configured target time. However,
+    the mechanism was not robust enough as it was based on the stats of the WAL
+    file in the Barman host (more specifically the creation time). For example:
+    if there were archiving or streaming lag between Postgres and Barman, that
+    could be enough for recovery to fail because Barman would miss to copy all
+    the required WAL files due to the weak check based on file stats.
+
+  - Pin `python-snappy` to `0.6.1` when running Barman through Python 3.6 or
+  older. Newer versions of `python-snappy` require `cramjam` on `2.7.0` or
+  newer, and these are only available for Python 3.7 or newer.
+
+  - `barman receive-wal` now exits with code `1` instead of `0` in the following
+    cases:
+
+    - Being unable to run with `--reset` flag because `pg_receivewal` is
+      running.
+
+    - Being unable to start `pg_receivewal` process because it is already
+      running.
+
+  - Fix and improve information about Python in `barman diagnose` output:
+
+    - The command now makes sure to use the correct Python interpreter where
+      Barman is installed when outputting the Python version through
+      `python_ver` JSON key. Previously, if a environment had multiple Python
+      installations and/or virtual environments, the output could be misleading.
```

Comment:
The first part is confusing. Which is the correct Python interpreter?

Reply:
The same one that is is being used by Barman. Let me rephrase this a bit.